### DISCUSSION TITLE
onChange method returned by "register(...)" is not being called

### DISCUSSION THREAD

**Discussion by ChristianHuff-DEV** [react-hook-form/react-hook-form#8960] [Q&A] [answered]
Example: https://codesandbox.io/s/onchange-1wg265

I have a custom input component which expects to receive `onChange: ChangeEventHandler<HTMLInputElement>;` in its `props`. 

I then add this `onChange` method to the input element (inside the `TextInput` component) (The `console.log(...)` is just for demonstration.):
```ts
          <input
            ref={ref}
            type="text"
            onChange={(event) => {
              console.log("TextInput.onChange");
              props.onChange(event);
            }}
            onBlur={props.onBlur}
            defaultValue={props.defaultValue}
            className="max-w-lg block w-full shadow-sm focus:ring-indigo-500 focus:border-indigo-500 text-sm border-gray-300 rounded-md"
          />
```
In the form I use `register(...)` to hook it up with *react-hook-form* as follows:
```ts
        <TextInput
          label="Name"
          {...register("name", {
            onChange: (event) => {
              console.log("App.onChange");
            },
            required: { value: true, message: "Name must be filled" }
          })}
          defaultValue="Peter"
          error={formState.errors.name?.message}
        />
```
The issue is that only the `onChange` method inside the custom `TextInput` component is called. The method I defined as part of the `register(...)` method is not called.

I tried removing the `onChange` method from the `TextInput` component. But it didn't help either.

From other discussion regarding issues with `onChange` it appears that often the problem is that the `onChange` method returned by `register(...)` is overriden. But from my understanding this isn't my issue since I'm using the `onChange` method of the `register(...)` function.

**Comment by bluebill1049** [comment]
share a codesandbox? you can attach custom `onChange` with register API as well

**Reply by Moshyfawn** [reply]
They have a csb; and it does act weird, I haven't looked into it yet
https://codesandbox.io/s/onchange-1wg265

**Comment by bluebill1049** [comment, marked as answer]
missing name prop: https://codesandbox.io/s/onchange-forked-x9giqj?file=/src/TextInput.tsx cc @Moshyfawn

**Reply by Moshyfawn** [reply]
Oooh, you are the most attentive person lol I'd sit and search for a missing `prop` for ages xD

**Reply by ChristianHuff-DEV** [reply]
Thank you!
